In [1]:
import pandas as pd
from IPython.display import clear_output
import numpy as np

# Applying DOI and MagIC Link search functions

In the case that you have many literature references that include a title, author, year, journal, volume, and pages but do not have the DOI
or MagIC Earthref Data DOI contribution link these functions can assist in finding these missing identifiers. 

Many are familiar with the reference DOI assigned to most published papers. The format we will make use of here will start with the 10 ('10.1029/A1B2C3'). 
We are interested in the MagIC link that relates to the rock magnetic data so that one can easily find the data associated with a paper. The http links produced by these functions are the Earthref Data DOI links related to contributions (http://dx.doi.org/10.7288/V4/MAGIC/19904).

These functions used here in this notebook are found in the module "dl_search". 

In [2]:
import dl_search as srch   # searching functions

## Functions and their uses
See the documentation contained in the dl_search.py file or using help()

In [3]:
help( srch.magic_link_from_doi )

Help on function magic_link_from_doi in module dl_search:

magic_link_from_doi(doi)
    This uses the earthref MagIC api to search for a magic contribution using a doi. 
    Adapted from pmagpy's "ipmag.download_magic_from_doi()"
    
    Input: 
        doi: str beginning with '10.'
        
    Output: 
        earthref_doi_link: http link if found
                           NaN if not found in MagIC database, try using magic_link_from_title to search the title in MagIC



### magic_link_from_doi 

In [4]:
doi = '10.1029/2021GC009990'
srch.magic_link_from_doi(doi)

'http://dx.doi.org/10.7288/V4/MAGIC/17452'

In [5]:
doi = '10.1029/2019GC008728'
srch.magic_link_from_doi(doi)

'http://dx.doi.org/10.7288/V4/MAGIC/16709'

### magic_link_from_title

In [6]:
title = "Geomagnetic field intensity between 70 000 and 130 000 years B.P. from a volcanic sequence on La Réunion, Indian Ocean"
#id = '19405'

srch.magic_link_from_title(title)

'http://dx.doi.org/10.7288/V4/MAGIC/19405'

### get_doi
Examples

In [7]:
title = 'Geomagnetic field intensity between 70 000 and 130 000 years B.P. from a volcanic sequence on La Reunion, Indian Ocean.' 
srch.get_doi(title)

'10.1016/0012-821x(96)00024-6'

### These can be used in tandem when magic_link_from_title cannot find the associated link

In [8]:
title = 'Evidence of anomalously weak geomagnetic field during Matuyama reversed epoch'
srch.magic_link_from_title(title)   # no content found

nan

The above does not work. So then we will try:

In [9]:
title = 'Evidence of anomalously weak geomagnetic field during Matuyama reversed epoch'
srch.magic_link_from_title2(title)

'http://dx.doi.org/10.7288/V4/MAGIC/18775'

Which is succesful.

## Working Example: Adding DOIs and Earthref Data DOI links to a spreadsheet of references of \*paper titles

\*Paper in the case of this file, may inclue pre-finalized paper titles.

In [10]:
# import references file with missing DOIs and MagIC links
pint_refs = pd.read_excel('../testdata/PINT_References_magiclinks.xlsx')
pint_refs_dois = pint_refs.loc[pint_refs['DOI'].isna() == False]     # collect data rows that have DOIs 

# print dataframe with this selection
pd.set_option('display.max_rows', 5)
print(len(pint_refs))
pint_refs_dois    # 36 have DOIs out of the 413 rows of references, we'll keep this for later

413


REFNO                                            AUTHORS  YEAR  \
375    745                      Abdulghafur, F., Bowles, J.A.  2019   
376    746  Sánchez-Moreno, E.M., Calvo-Rathert, M., Gogui...  2019   
..     ...                                                ...   ...   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
375  Absolute Paleointensity Study of Miocene Tiva ...   
376  Weak palaeointensity results over a Pliocene v...   
..                                                 ...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                  JOURNAL  VOL          PAGES  \
375  Geochemistry, Geophysics, Geosystems   20      5818–5830   
376    Geophysical Journal International   220      1604–1618   
..                                    ...  ...            ...   
411  Geochemistry, Geophysics, Geosystems   23  e2022GC010473   
412  Geochemistry, Geophysics, Geosystems   22  e2020GC009605   

                                      DOI MagIC  
375  https://doi.org/10.1029/2019GC008728   NaN  
376    https://doi.org/10.1093/gji/ggz533   NaN  
..                                    ...   ...  
411  https://doi.org/10.1029/2022GC010473   NaN  
412  https://doi.org/10.1029/2020GC009605   NaN  

[36 rows x 9 columns]

In [11]:
pint_refs # with the section of the table with missing DOIs only
pint_refs_titles = pint_refs['TITLE']    # assign colums to variables

dois = []    # container for DOIs found
for i in range(len(pint_refs_titles)):   # for each index of the number of references (413)
    dois.append(srch.get_doi(pint_refs_titles[i]))   # apply get_doi
    clear_output(wait=True)
    print( i+1,'/{n:5d}'.format(n=len(pint_refs_titles)) )   # prints progress, takes ~5 min to run

413 /  413


In [12]:
dois # view the list of DOIs
# set list of DOIs to a new name
dois_n = dois
dois_n[0]

'10.5636/jgg.23.129'

In [13]:
# fill a new column in our dataframe with DOIs found
pint_refs['DOI_found'] = dois_n

# view
pd.set_option('display.max_rows', 10)
pint_refs

REFNO                                            AUTHORS  YEAR  \
0        1   Aoki, Y., Kase, H., Ishibashi, K., Kinoshita, H.  1971   
1        2        Bagina, O.L., Minasyan, D.O., Petrova, G.N.  1976   
2        3                                        Bergh, H.W.  1970   
3        4                             Bogue, S.W., Coe, R.S.  1984   
4        5  Bol'shakov, A.S., Solodovnikov, G.M., Vechfins...  1978   
..     ...                                                ...   ...   
408    779  Lloyd, S. J., Biggin, A. J., Paterson, G. A., ...  2022   
409    780  Thallner, D., Shcherbakova, V. V., Bakhmutov, ...  2022   
410    781                  Biasi, J., Kirschvink, J., Fu, R.  2021   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
0    Evidence of anomalously weak geomagnetic field...   
1    Determination of the intensity of the ancient ...   
2    Paleomagnetism of the Stillwater Complex, Montana   
3    Transitional paleointensities from Kauai, Hawa...   
4    Determination of the geomagnetic field strengt...   
..                                                 ...   
408  Extremely weak early Cambrian dipole moment si...   
409  New palaeodirections and palaeointensity data ...   
410  Characterizing the Geomagnetic Field at High S...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                               JOURNAL      VOL  \
0                                 J. Geomag. Geoelect.       23   
1                        Izv. Akad. Nauk. (in Russian)        2   
2    Paleogeophysics (ed.  S.K. Runcorn, Academic P...       17   
3                                     J. Geophys. Res.       89   
4                      Izv., Earth Phys. (Eng. Trans.)       14   
..                                                 ...      ...   
408                Earth and Planetary Science Letters      595   
409                                   Geophys. J. Int.  231 (1)   
410                                   J. Geophys. Res.      126   
411               Geochemistry, Geophysics, Geosystems       23   
412               Geochemistry, Geophysics, Geosystems       22   

             PAGES                                         DOI MagIC  \
0          129-132                                         NaN   NaN   
1            81-86                                         NaN   NaN   
2          143-158                                         NaN   NaN   
3      10341-10354                                         NaN   NaN   
4          904-910                                         NaN   NaN   
..             ...                                         ...   ...   
408         117757  https://doi.org/10.1016/j.epsl.2022.117757   NaN   
409        474-492         https://doi.org/10.1093/gji/ggac186   NaN   
410  e2021JB023273        https://doi.org/10.1029/2021JB023273   NaN   
411  e2022GC010473        https://doi.org/10.1029/2022GC010473   NaN   
412  e2020GC009605        https://doi.org/10.1029/2020GC009605   NaN   

                      DOI_found  
0            10.5636/jgg.23.129  
1                           NaN  
2                           NaN  
3       10.1029/jb089ib12p10341  
4                           NaN  
..                          ...  
408  10.1016/j.epsl.2022.117757  
409         10.1093/gji/ggac186  
410        10.1029/2021jb023273  
411   10.1002/essoar.10511070.2  
412        10.1029/2020gc009605  

[413 rows x 10 columns]

In [14]:
# now to find the MagIC Links
magic_links = []     # MagIC link container
for i in range(len(dois_n)): 
    magic_links.append(srch.magic_link_from_doi(dois_n[i]))
    clear_output(wait=True)
    print( i+1,'/{n:5d}'.format(n=len(dois_n)) )   # prints progress, takes ~5 min to run

413 /  413


In [15]:
magic_links  # view the list of MagIC link

# set list to a new name
pint_refs['MagIC_found'] = magic_links
pint_refs

REFNO                                            AUTHORS  YEAR  \
0        1   Aoki, Y., Kase, H., Ishibashi, K., Kinoshita, H.  1971   
1        2        Bagina, O.L., Minasyan, D.O., Petrova, G.N.  1976   
2        3                                        Bergh, H.W.  1970   
3        4                             Bogue, S.W., Coe, R.S.  1984   
4        5  Bol'shakov, A.S., Solodovnikov, G.M., Vechfins...  1978   
..     ...                                                ...   ...   
408    779  Lloyd, S. J., Biggin, A. J., Paterson, G. A., ...  2022   
409    780  Thallner, D., Shcherbakova, V. V., Bakhmutov, ...  2022   
410    781                  Biasi, J., Kirschvink, J., Fu, R.  2021   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
0    Evidence of anomalously weak geomagnetic field...   
1    Determination of the intensity of the ancient ...   
2    Paleomagnetism of the Stillwater Complex, Montana   
3    Transitional paleointensities from Kauai, Hawa...   
4    Determination of the geomagnetic field strengt...   
..                                                 ...   
408  Extremely weak early Cambrian dipole moment si...   
409  New palaeodirections and palaeointensity data ...   
410  Characterizing the Geomagnetic Field at High S...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                               JOURNAL      VOL  \
0                                 J. Geomag. Geoelect.       23   
1                        Izv. Akad. Nauk. (in Russian)        2   
2    Paleogeophysics (ed.  S.K. Runcorn, Academic P...       17   
3                                     J. Geophys. Res.       89   
4                      Izv., Earth Phys. (Eng. Trans.)       14   
..                                                 ...      ...   
408                Earth and Planetary Science Letters      595   
409                                   Geophys. J. Int.  231 (1)   
410                                   J. Geophys. Res.      126   
411               Geochemistry, Geophysics, Geosystems       23   
412               Geochemistry, Geophysics, Geosystems       22   

             PAGES                                         DOI MagIC  \
0          129-132                                         NaN   NaN   
1            81-86                                         NaN   NaN   
2          143-158                                         NaN   NaN   
3      10341-10354                                         NaN   NaN   
4          904-910                                         NaN   NaN   
..             ...                                         ...   ...   
408         117757  https://doi.org/10.1016/j.epsl.2022.117757   NaN   
409        474-492         https://doi.org/10.1093/gji/ggac186   NaN   
410  e2021JB023273        https://doi.org/10.1029/2021JB023273   NaN   
411  e2022GC010473        https://doi.org/10.1029/2022GC010473   NaN   
412  e2020GC009605        https://doi.org/10.1029/2020GC009605   NaN   

                      DOI_found                               MagIC_found  
0            10.5636/jgg.23.129  http://dx.doi.org/10.7288/V4/MAGIC/18775  
1                           NaN                                       NaN  
2                           NaN                                       NaN  
3       10.1029/jb089ib12p10341  http://dx.doi.org/10.7288/V4/MAGIC/18753  
4                           NaN                                       NaN  
..                          ...                                       ...  
408  10.1016/j.epsl.2022.117757  http://dx.doi.org/10.7288/V4/MAGIC/19529  
409         10.1093/gji/ggac186  http://dx.doi.org/10.7288/V4/MAGIC/19341  
410        10.1029/2021jb023273  http://dx.doi.org/10.7288/V4/MAGIC/19313  
411   10.1002/essoar.1051

### Update references with initial DOIs
Now with the dataframe from earlier, with some found DOIs, we'll search for their corresponding links to their MagIC contribution. We can apply the *magic_link_from_doi* function to the part of our data that had DOIs already. 

In [16]:
print( len(pint_refs_dois) )
pint_refs_dois

36


REFNO                                            AUTHORS  YEAR  \
375    745                      Abdulghafur, F., Bowles, J.A.  2019   
376    746  Sánchez-Moreno, E.M., Calvo-Rathert, M., Gogui...  2019   
377    747  Cervantes-Solano, M., Goguitchaichvili, A., Sá...  2020   
378    749  Miki, M., Seki, H., Yamamoto, Y., Gouzu, C., H...  2020   
379    750  Shcherbakova, V.V., Bakhmutov, V.G., Thallner,...  2020   
..     ...                                                ...   ...   
408    779  Lloyd, S. J., Biggin, A. J., Paterson, G. A., ...  2022   
409    780  Thallner, D., Shcherbakova, V. V., Bakhmutov, ...  2022   
410    781                  Biasi, J., Kirschvink, J., Fu, R.  2021   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
375  Absolute Paleointensity Study of Miocene Tiva ...   
376  Weak palaeointensity results over a Pliocene v...   
377  An integrated paleomagnetic and multispecimen ...   
378  Paleomagnetism, paleointensity and geochronolo...   
379  Ultra-low palaeointensities from East European...   
..                                                 ...   
408  Extremely weak early Cambrian dipole moment si...   
409  New palaeodirections and palaeointensity data ...   
410  Characterizing the Geomagnetic Field at High S...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                      JOURNAL      VOL          PAGES  \
375      Geochemistry, Geophysics, Geosystems       20      5818–5830   
376        Geophysical Journal International       220      1604–1618   
377  Journal of South American Earth Sciences      104         102815   
378                    Journal of Geodynamics      139         101752   
379         Geophysical Journal International      220      1928-1946   
..                                        ...      ...            ...   
408       Earth and Planetary Science Letters      595         117757   
409                          Geophys. J. Int.  231 (1)        474-492   
410                          J. Geophys. Res.      126  e2021JB023273   
411      Geochemistry, Geophysics, Geosystems       23  e2022GC010473   
412      Geochemistry, Geophysics, Geosystems       22  e2020GC009605   

                                              DOI MagIC  
375          https://doi.org/10.1029/2019GC008728   NaN  
376            https://doi.org/10.1093/gji/ggz533   NaN  
377  https://doi.org/10.1016/j.jsames.2020.102815   NaN  
378     https://doi.org/10.1016/j.jog.2020.101752   NaN  
379            https://doi.org/10.1093/gji/ggz566   NaN  
..                                            ...   ...  
408    https://doi.org/10.1016/j.epsl.2022.117757   NaN  
409           https://doi.org/10.1093/gji/ggac186   NaN  
410          https://doi.org/10.1029/2021JB023273   NaN  
411          https://doi.org/10.1029/2022GC010473   NaN  
412          https://doi.org/10.1029/2020GC009605   NaN  

[36 rows x 9 columns]

For the 36 PINT references that already have a listed DOI, some have the "doi.org" attached to it but these functions only read in the number ID. The code below secludes the DOI number. 

In [17]:
# seclude doi number as a str
df_dois = pint_refs_dois['DOI'].values
doi_vals = []
for i in range(len(df_dois)): 
    if len( df_dois[i].split('.org/') ) == 2:   # if the column value has doi.org + id number 
        doi_vals.append(df_dois[i].split('.org/')[1])
    elif len( df_dois[i].split('.org/') ) == 1:  # if the column value has just the id number
        doi_vals.append(df_dois[i].split('.org/')[0])

In [18]:
#doi_vals # check the values 

Now with our array with everything from our "DOI" column in the same format, we can run the search function.

In [19]:
# send DOI number to be searched in MagIC
magic_links = []
for i in range(len(doi_vals)): 
    magic_links.append(srch.magic_link_from_doi(doi_vals[i])) # takes ~5 mins

In [20]:
pint_refs_dois['MagIC_found'] = magic_links  # create a new column for our new MagIC links
pint_refs_dois  # no need to worry about the warning for now 

C:\Users\trin\AppData\Local\Temp/ipykernel_16276/1004045502.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pint_refs_dois['MagIC_found'] = magic_links  # create a new column for our new MagIC links


REFNO                                            AUTHORS  YEAR  \
375    745                      Abdulghafur, F., Bowles, J.A.  2019   
376    746  Sánchez-Moreno, E.M., Calvo-Rathert, M., Gogui...  2019   
377    747  Cervantes-Solano, M., Goguitchaichvili, A., Sá...  2020   
378    749  Miki, M., Seki, H., Yamamoto, Y., Gouzu, C., H...  2020   
379    750  Shcherbakova, V.V., Bakhmutov, V.G., Thallner,...  2020   
..     ...                                                ...   ...   
408    779  Lloyd, S. J., Biggin, A. J., Paterson, G. A., ...  2022   
409    780  Thallner, D., Shcherbakova, V. V., Bakhmutov, ...  2022   
410    781                  Biasi, J., Kirschvink, J., Fu, R.  2021   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
375  Absolute Paleointensity Study of Miocene Tiva ...   
376  Weak palaeointensity results over a Pliocene v...   
377  An integrated paleomagnetic and multispecimen ...   
378  Paleomagnetism, paleointensity and geochronolo...   
379  Ultra-low palaeointensities from East European...   
..                                                 ...   
408  Extremely weak early Cambrian dipole moment si...   
409  New palaeodirections and palaeointensity data ...   
410  Characterizing the Geomagnetic Field at High S...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                      JOURNAL      VOL          PAGES  \
375      Geochemistry, Geophysics, Geosystems       20      5818–5830   
376        Geophysical Journal International       220      1604–1618   
377  Journal of South American Earth Sciences      104         102815   
378                    Journal of Geodynamics      139         101752   
379         Geophysical Journal International      220      1928-1946   
..                                        ...      ...            ...   
408       Earth and Planetary Science Letters      595         117757   
409                          Geophys. J. Int.  231 (1)        474-492   
410                          J. Geophys. Res.      126  e2021JB023273   
411      Geochemistry, Geophysics, Geosystems       23  e2022GC010473   
412      Geochemistry, Geophysics, Geosystems       22  e2020GC009605   

                                              DOI MagIC  \
375          https://doi.org/10.1029/2019GC008728   NaN   
376            https://doi.org/10.1093/gji/ggz533   NaN   
377  https://doi.org/10.1016/j.jsames.2020.102815   NaN   
378     https://doi.org/10.1016/j.jog.2020.101752   NaN   
379            https://doi.org/10.1093/gji/ggz566   NaN   
..                                            ...   ...   
408    https://doi.org/10.1016/j.epsl.2022.117757   NaN   
409           https://doi.org/10.1093/gji/ggac186   NaN   
410          https://doi.org/10.1029/2021JB023273   NaN   
411          https://doi.org/10.1029/2022GC010473   NaN   
412          https://doi.org/10.1029/2020GC009605   NaN   

                                  MagIC_found  
375  http://dx.doi.org/10.7288/V4/MAGIC/16709  
376  http://dx.doi.org/10.7288/V4/MAGIC/17131  
377                                       NaN  
378                                       NaN  
379  http://dx.doi.org/10.7288/V4/MAGIC/16733  
..                                        ...  
408  http://dx.doi.org/10.7288/V4/MAGIC/19529  
409  http://dx.doi.org/10.7288/V4/MAGIC/19341  
410  http://dx.doi.org/10.7288/V4/MAGIC/19313  
411  http://dx.doi.org/10.7288/V4/MAGIC/19491  
412  http://dx.doi.org/10.7288/V4/MAGIC/16869  

[36 rows x 10 columns]

Now to combine our two partial dataframes back into one dataframe ready for export

In [21]:
# combining the two dataframes
pint_refs_links_filled = pd.concat([pint_refs, pint_refs_dois])
pint_refs_links_filled

REFNO                                            AUTHORS  YEAR  \
0        1   Aoki, Y., Kase, H., Ishibashi, K., Kinoshita, H.  1971   
1        2        Bagina, O.L., Minasyan, D.O., Petrova, G.N.  1976   
2        3                                        Bergh, H.W.  1970   
3        4                             Bogue, S.W., Coe, R.S.  1984   
4        5  Bol'shakov, A.S., Solodovnikov, G.M., Vechfins...  1978   
..     ...                                                ...   ...   
408    779  Lloyd, S. J., Biggin, A. J., Paterson, G. A., ...  2022   
409    780  Thallner, D., Shcherbakova, V. V., Bakhmutov, ...  2022   
410    781                  Biasi, J., Kirschvink, J., Fu, R.  2021   
411    782  Tauxe, L., Asefaw, H., Behar, N., Koppers, A. ...  2022   
412    783  di Chiara, A., Tauxe, L., Staudigel, H., Flori...  2021   

                                                 TITLE  \
0    Evidence of anomalously weak geomagnetic field...   
1    Determination of the intensity of the ancient ...   
2    Paleomagnetism of the Stillwater Complex, Montana   
3    Transitional paleointensities from Kauai, Hawa...   
4    Determination of the geomagnetic field strengt...   
..                                                 ...   
408  Extremely weak early Cambrian dipole moment si...   
409  New palaeodirections and palaeointensity data ...   
410  Characterizing the Geomagnetic Field at High S...   
411  Paleointensity Estimates From the Pleistocene ...   
412  Earth's Magnetic Field Strength and the Cretac...   

                                               JOURNAL      VOL  \
0                                 J. Geomag. Geoelect.       23   
1                        Izv. Akad. Nauk. (in Russian)        2   
2    Paleogeophysics (ed.  S.K. Runcorn, Academic P...       17   
3                                     J. Geophys. Res.       89   
4                      Izv., Earth Phys. (Eng. Trans.)       14   
..                                                 ...      ...   
408                Earth and Planetary Science Letters      595   
409                                   Geophys. J. Int.  231 (1)   
410                                   J. Geophys. Res.      126   
411               Geochemistry, Geophysics, Geosystems       23   
412               Geochemistry, Geophysics, Geosystems       22   

             PAGES                                         DOI MagIC  \
0          129-132                                         NaN   NaN   
1            81-86                                         NaN   NaN   
2          143-158                                         NaN   NaN   
3      10341-10354                                         NaN   NaN   
4          904-910                                         NaN   NaN   
..             ...                                         ...   ...   
408         117757  https://doi.org/10.1016/j.epsl.2022.117757   NaN   
409        474-492         https://doi.org/10.1093/gji/ggac186   NaN   
410  e2021JB023273        https://doi.org/10.1029/2021JB023273   NaN   
411  e2022GC010473        https://doi.org/10.1029/2022GC010473   NaN   
412  e2020GC009605        https://doi.org/10.1029/2020GC009605   NaN   

                   DOI_found                               MagIC_found  
0         10.5636/jgg.23.129  http://dx.doi.org/10.7288/V4/MAGIC/18775  
1                        NaN                                       NaN  
2                        NaN                                       NaN  
3    10.1029/jb089ib12p10341  http://dx.doi.org/10.7288/V4/MAGIC/18753  
4                        NaN                                       NaN  
..                       ...                                       ...  
408                      NaN  http://dx.doi.org/10.7288/V4/MAGIC/19529  
409                      NaN  http://dx.doi.org/10.7288/V4/MAGIC/19341  
410                      NaN  http://dx.doi.org/10.7288/V4/MAGIC/19313  
411                      NaN  http://dx.doi.org/10.7288

Depending on the paper titles included in our original spreadsheet (some errors may include: spelling, if they differ from the published paper title, if they aren't in MagIC, no assigned Earthref Data DOI etc.), the DOI and MagIC search results may vary. 

This is a summary of our results from this test case:

Out of our 413 refrences, we started off with: 
- 36 DOI links or IDs
- 0 working MagIC Earthref Data DOI links

In [22]:
print( pint_refs_links_filled['DOI_found'].describe() )
print( pint_refs_links_filled['MagIC_found'].describe() )

count                  356
unique                 355
top       10.1360/02yd9092
freq                     2
Name: DOI_found, dtype: object
count                                          241
unique                                         223
top       http://dx.doi.org/10.7288/V4/MAGIC/17131
freq                                             2
Name: MagIC_found, dtype: object


This tells us we now have: 
- 356 DOI ID links (86% with 57 NaN) 
- 224 MagIC EarthRef Data DOI links (54% with 208 NaN)

To export this file as an excel sheet into our 'testdata' folder: 

In [24]:
pint_refs_links_filled.to_excel('../testdata/PINT_references_ex.xlsx', index=False)